subprojects/docs/src/docs/userguide/publishingMaven.adoc

```diff
@@ -300,3 +300,48 @@ The result is that the following artifacts will be published:
 * The Javadoc JAR artifact that has been explicitly configured: `my-library-1.0-javadoc.jar`
 
 The <<signing_plugin, Signing Plugin>> is used to generate a signature file for each artifact. In addition, checksum files will be generated for all artifacts and signature files.
+
+[[publishing_maven:deferred_configuration]]
+=== Removal of deferred configuration behavior
+
+Until Gradle 4.7, the `publishing {}` block was implicitly treated as if all the logic inside it was executed after the project is evaluated.
+This caused quite a bit of confusion, because it was the only block that behaved that way.
+As part of the stabilization effort in Gradle 4.8, we are deprecating this behavior and asking all users to migrate their build.
+
+The new, stable behavior can be switched on by adding the following to your settings file:
+
+    enableFeaturePreview('STABLE_PUBLISHING')
+
+We recommend doing a test run with a local repository to see whether all artifacts still have the expected coordinates.
+In most cases everything should work as before and you are done.
+
+If the coordinates change unexpectedly, you may have some logic inside your publishing block or in a plugin that is depending on the deferred configuration behavior.
+For instance, the following logic assumes that the subprojects will be evaluated when the artifactId is set:
+
+    subprojects {
+        publishing {
+            publications {
+                mavenJava {
+                    from components.java
+                    artifactId = jar.baseName
+                }
+            }
+        }
+    }
+
+This kind of logic must be wrapped in an `afterEvaluate {}` block to make it work going forward.
+
+    subprojects {
+        publishing {
+            publications {
+                mavenJava {
+                    from components.java
+                    afterEvaluate {
+                        artifactId = jar.baseName
+                    }
+                }
+            }
+        }
+    }
+
+Gradle 5.0 will switch that feature on by default, so please make sure your build is compatible as soon as possible.
```

Comment:
Should this section also be in `publishingIvy.adoc` and, if so, where do we link to from the warning printed by the common `publishing` plugin?

Reply:
I think having it here is fine. Ideally it would live in an overview doc, but it's probably not a big deal.

Reply:
Information like this should be in a callout of some sort. It's important and I assume you want to draw users' attention to it. It's likely to be missed when it's at the end of a section like this in a single-sentence paragraph. Many readers will skim past it.

Reply:
Done